graph/client.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import print_function, unicode_literals
+import re
+import graph.xml_parser
+import graph.oval_graph
+import webbrowser
+import json
+import argparse
+
+
+class client():
+    def __init__(self, args):
+        self.arg = self.parse_arguments(args)
+        self.source_filename = self.arg.source_filename
+        self.rule_name = self.arg.rule_id
+        self.xml_parser = graph.xml_parser.xml_parser(self.source_filename)
+
+    def run_gui_and_return_answers(self):
+        try:
+            from PyInquirer import style_from_dict, Token, prompt, Separator
+            return prompt(self.get_questions(Separator('= The Rules ID =')))
+        except ImportError:
+            print('== The Rules ID ==')
+            for rule in self.search_rules_id():
+                print(rule['id_rule'] + r'\b')
+            return None
+
+    def get_questions(self, separator):
+        rules = self.search_rules_id()
+        questions = [{
+            'type': 'checkbox',
+            'message': 'Select rule(s)',
+            'name': 'rules',
+            'choices': [
+                separator],
+            'validate': (lambda answer: 'You must choose at least one topping.'
+                         if len(answer) == 0 else True)
+        }]
+        for rule in rules:
+            questions[0]['choices'].append(dict(name=rule['id_rule']))
+        return questions
+
+    def _get_wanted_rules(self):
+        return [
+            x for x in self.xml_parser.get_used_rules() if re.search(
+                self.rule_name, x['id_rule'])]
+
+    def _get_wanted_not_selected_rules(self):
+        return [
+            x for x in self.xml_parser.get_notselected_rules() if re.search(
+                self.rule_name, x['id_rule'])]
+
+    def search_rules_id(self):
+        rules = self._get_wanted_rules()
+        notselected_rules = self._get_wanted_not_selected_rules()
+        if len(notselected_rules) and not rules:
+            raise ValueError(
+                'err- rule(s) "{}" was not selected, so there are no results.'
```

Comment:
I would expand it. The rule is "notselected"  because it wasn't a part of the executed profile and therefore it wasn't evaluated during the scan.